src/common/chains.ts

```diff
@@ -9,3 +11,383 @@ export enum SupportedChainId {
   BASE = 8453,
   SEPOLIA = 11155111,
 }
+
+/**
+ * The chain id of the chain.
+ *
+ * TODO: Should we generalize it even more to allow non-EVM chains? We should probably revisit also the chain interface, and some other types.
+ */
+export type ChainId = number
```

Comment:
Yes.

How about:
type EvmChainId = number
type NonEvmChainId = string
export type ChainId = EvmChainId | NonEvmChainId

Does that complicates things too much?
We could make everything a string to reduce the confusion.

But we can also leave it be for now.

Reply:
Because we don't support yet non-evm chains and still don't know how we will do it, I think is better to delay this. 

This chain will be compatible with adding the NonEvmChainId type